test/dotnet-list-proj.Tests/GivenDotnetListProj.cs

```diff
@@ -0,0 +1,141 @@
+            cmd.Should().Fail();
+            cmd.StdErr.Should().Contain("Could not find");
+            cmd.StdOut.Should().Contain("Usage:");
+        }
+
+        [Fact]
+        public void WhenInvalidSolutionIsPassedItPrintsErrorAndUsage()
+        {
+            var projectDirectory = TestAssetsManager.CreateTestInstance("InvalidSolution")
+                                                    .WithLockFiles()
+                                                    .Path;
+
+            var cmd = new DotnetCommand()
+                .WithWorkingDirectory(projectDirectory)
+                .ExecuteWithCapturedOutput("list InvalidSolution.sln projects");
+            cmd.Should().Fail();
+            cmd.StdErr.Should().Contain("Invalid solution ");
+            cmd.StdOut.Should().Contain("Usage:");
+        }
+
+        [Fact]
+        public void WhenInvalidSolutionIsFoundItPrintsErrorAndUsage()
+        {
+            var projectDirectory = TestAssetsManager.CreateTestInstance("InvalidSolution")
+                                                    .WithLockFiles()
+                                                    .Path;
+
+            var cmd = new DotnetCommand()
+                .WithWorkingDirectory(projectDirectory)
+                .ExecuteWithCapturedOutput("list projects");
+            cmd.Should().Fail();
+            cmd.StdErr.Should().Contain("Invalid solution ");
+            cmd.StdOut.Should().Contain("Usage:");
+        }
+
+        [Fact]
+        public void WhenNoSolutionExistsInTheDirectoryItPrintsErrorAndUsage()
+        {
+            var projectDirectory = TestAssetsManager.CreateTestInstance("TestAppWithSlnAndCsprojFiles")
+                                                    .WithLockFiles()
+                                                    .Path;
+
+            var cmd = new DotnetCommand()
+                .WithWorkingDirectory(Path.Combine(projectDirectory, "App"))
+                .ExecuteWithCapturedOutput("list projects");
+            cmd.Should().Fail();
+            cmd.StdErr.Should().Contain("does not exist");
+            cmd.StdOut.Should().Contain("Usage:");
+        }
+
+        [Fact]
+        public void WhenMoreThanOneSolutionExistsInTheDirectoryItPrintsErrorAndUsage()
+        {
+            var projectDirectory = TestAssetsManager.CreateTestInstance("TestAppWithMultipleSlnFiles")
+                                                    .WithLockFiles()
+                                                    .Path;
+
+            var cmd = new DotnetCommand()
+                .WithWorkingDirectory(projectDirectory)
+                .ExecuteWithCapturedOutput("list projects");
+            cmd.Should().Fail();
+            cmd.StdErr.Should().Contain("more than one");
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Fact]
+        public void WhenNoProjectReferencesArePresentInTheSolutionItPrintsError()
```

Comment:
I would update the name of this test slightly. You say it should print an error, which gives me an impression that the command should fail. However, we are asserting that the command should pass. Maybe say ItPrintsANoProjectMessage or something like that.